server/src/main/java/com/cloud/network/NetworkModelImpl.java

```diff
@@ -1659,8 +1659,7 @@ public void checkNetworkPermissions(Account owner, Network network) {
             throw new CloudRuntimeException("cannot check permissions on (Network) <null>");
         }
         // Perform account permission check
-        if ((network.getGuestType() != GuestType.Shared && network.getGuestType() != GuestType.L2) ||
-                (network.getGuestType() == GuestType.Shared && network.getAclType() == ACLType.Account)) {
+        if (network.getGuestType() != GuestType.Shared || network.getAclType() == ACLType.Account) {
```

Comment:
@weizhouapache check the changes introduced at https://github.com/apache/cloudstack/pull/2420, for L2 network.